web/src/test/java/org/springframework/security/web/firewall/StrictHttpFirewallTests.java

```diff
@@ -731,6 +759,21 @@ public void getFirewalledRequestGetHeaderWhenNotAllowedHeaderValueThenException(
 		assertThatExceptionOfType(RequestRejectedException.class).isThrownBy(() -> request.getHeader("good name"));
 	}
 
+	@Test
+	public void getFirewalledRequestGetHeaderWhenSetNotAllowedHeaderValuesThenException() {
+		this.request.addHeader("good name", "bad value");
+		this.request.addHeader("best name", "worst value");
+
+		this.firewall.setAllowedHeaderValues((value) -> !value.equals("bad value"));
```

Comment:
Since the second call overrides the first, you can leave this line out and it won't change the result of the test.

Reply:
Thank you for the review.

I have updated the code to retain only the first call.